### PR TITLE
DRAFT: Add modules case-study

### PR DESCRIPTION
**Pull request by aornugent**
This is a markdown version of Shubham's EMO report on FLINT Modules. It is still a work in progress, but we'd like to try using GitHub for reviews. 

There may end up being material here for several smaller case-studies that can be developed during GSoD.

### CODE REVIEW

File: case-studies/modules.md [hunk 1]
```diff
@@ -0,0 +1,1406 @@
...
+* [Developing modules](#developing-modules)
+* [What are the typical components of a module?](#module-components)
+* [How can existing modules be combined?](#combining-modules)
+* [How do I create a new module?](#creating-modules)
+* [Translating a system of equations into C++](#process-model)
+* [Wrapping existing models](#foriegn-functions)
+* [How do I make a point-based module spatially explicit?](#spatial-modules)
+* [Creating custom modules](#custom-modules)
+* [What are Disturbance Events?](#disturbance-events)
+* [Key Components of the FLINT](#flint-namespace)
```
Comment (aornugent) [inline case-studies/modules.md:25]: These links need to be formated `#modules.md-goal` to work properly. You can check the permalink in the GitHub preview.

**Comment by aornugent** [comment]
@shubhamkarande13 - can you please add a list of items in progress or to be added in the future for other contributors to pick up?

**Comment by shubhamkarande13** [comment]
@aornugent We already have a repository of [FLINT modules docs](https://github.com/moja-global/FLINT_Modules_docs). 
Should we continue on that repository or here?
I am currently making a list of the items in progress. I plan to get everything done by tomorrow! :)

**Comment by aornugent** [comment]
Sure, let's move it to there.

I'll leave this PR open for the Docusaurus PoC in #2.

**Pull request closed without merging**